src/rustup-utils/src/raw.rs

```diff
@@ -152,161 +154,84 @@ pub fn tee_file<W: io::Write>(path: &Path, mut w: &mut W) -> io::Result<()> {
     }
 }
 
-pub fn download_file<P: AsRef<Path>>(url: hyper::Url,
+pub fn download_file<P: AsRef<Path>>(url: &str,
```

Comment:
Can you instead switch to `url::Url` instead of `str` (it's already used instead of `hyper::Url` everywhere els)? I hope to push Urls upward through the APIs to improve type safety.